astrocut/make_cube.py

```diff
@@ -346,3 +349,500 @@ def make_cube(self, file_list, cube_file="img-cube.fits", sector=None, max_memor
+        # Loop through files
+        for i, fle in enumerate(self.file_list):
+
+            if verbose:
+                st = time()
+
+            with fits.open(fle, mode='denywrite', memmap=True) as ffi_data:
+
+                # add the image and info to the arrays
+                sub_cube[:, :, i, 0] = ffi_data[0].data[start_row:end_row, :]
+
+                del ffi_data[0].data
+               
+                if fill_info_table:  # Also save the header info in the info table
+
+                    for kwd in self.info_table.columns:  # Iterate over every keyword in the TICA FFI primary header
+                        if kwd == "FFI_FILE":
+                            self.info_table[kwd][i] = os.path.basename(fle)
+                            
+                        else:
+                            nulval = None
+                            if self.info_table[kwd].dtype.name == "int32":
+                                nulval = 0
+                            elif self.info_table[kwd].dtype.char == "S":  # hacky way to check if it's a string
+                                nulval = ""
+                            
+                            # NOTE: 
+                            # A header keyword ('COMMENT') in TICA returns a keyword 
+                            # value in the form of a _HeaderCommentaryCard instead of a STRING. 
+                            # This breaks the info table because the info table 
+                            # doesn't understand what a FITS header commentary card is. 
+                            # Adding a try/except is a way to catch when these instances happen 
+                            # and turn the keyword value from a _HeaderCommentaryCard to a
+                            # string, which is what it's meant to be in the info table. 
+                            #
+                            # TO-DO: Find a more elegant way to handle these stray 
+                            # _HeaderCommentaryCards. 
+                            
+                            try:
+                                self.info_table[kwd][i] = ffi_data[0].header.get(kwd, nulval)
+                            except ValueError:
+                                kwd_val = ffi_data[0].header.get(kwd)
+                                if isinstance(kwd_val, fits.header._HeaderCommentaryCards):
+                                    self.info_table[kwd][i] = str(kwd_val)
+
+            if verbose:
+                print(f"Completed file {i} in {time()-st:.3} sec.")
+
+        # Fill block and flush to disk
+        if not self.update:
+            cube_hdu[1].data[start_row:end_row, :, :, :] = sub_cube
+        else:
+            self.cube_append[start_row:end_row, :, :, :] = sub_cube
+
+        if (version_info <= (3, 8)) or (platform == "win32"):
+            cube_hdu.flush()
+
+        del sub_cube
+    
+    def _update_info_table(self):
+        """ Updating an existing info table with newly created
+        """
```

Comment:
Is there some way you can re-use the code in `_build_info_table()` here instead of creating another method with largely the same code?

Reply:
Definitely. I'll add this as a task for when I'm optimizing the code, but for this PR I'll leave it as is because the functionality works OK.